### ISSUE TITLE
Segfault while Performing All-to-All Collective Operation on 8xH100 SXM5 (Shard + Swapaxes + Shard)

### ISSUE THREAD

**Issue by novastar53** [jax-ml/jax#30335]
### Description

I am trying to implement an operation that involves sharding an array along the leading axis followed by swapping the leading axis with axis 2, and finally, resharding along the new leading axis. This operations works fine on most configurations. However, It doesn't appear to work on a particular hardware configuration - 8 x Nvidia H100 80GB SMX5. 

Here's a minimal example to reproduce the issue:

`import os

import jax
import jax.numpy as jnp

devices = jax.devices()
num_devices = len(devices)

from jax.sharding import Mesh, PartitionSpec, NamedSharding

mesh = Mesh(devices, ("devices",))
spec = PartitionSpec("devices",)
sharding = NamedSharding(mesh, spec)

x = jax.random.normal(jax.random.key(0), (num_devices, 2, num_devices, 3))
x = jax.lax.with_sharding_constraint(x, sharding)
x = jnp.swapaxes(x, 0, 2)
x = jnp.reshape(x, (-1, 3)) # Program crashes here
x = jax.lax.with_sharding_constraint(x, sharding)
x = jnp.reshape(x, (num_devices, 2*num_devices, 3))
`

### System info (python version, jaxlib version, accelerator, etc.)

Accelerator: 8 x Nvidia H100 80GB SMX5. 
OS: Ubuntu Linux
Hardware: x86_64

JAX version: 0.5.2
Flax version 0.10.6

NVIDIA-SMI 570.148.08            
Driver Version: 570.148.08     
CUDA Version: 12.8

**Comment by rdyro** [comment]
That's not great! 

Can you try putting the swapaxes and reshape under a single jit as a workaround? 

Can you also try a newer version of JAX to see if the error persists?

**Comment by rdyro** [comment]
I tried the same hardware configuration with `jax[cuda]==0.5.2` and Python 3.13.4, but could not reproduce this.

Were you able to avoid segfaults nuder a different JAX version?

**Comment by novastar53** [comment]
HI @rdyro! I successfully ran this program on a different cloud provider with the same GPU configuration (8xH100 SXM5). There might be some differences in the provider library/software versions. 

I can compare the configurations and get back to you if you'd like.

**Comment by rdyro** [comment]
If you can easily identify the differing configurations, that'd be very useful, thanks. Your snippet should definitely not be segfaulting!

**Comment by novastar53** [comment]
Hi @rdyro, so I ran the code snippet on the original cloud provider again. This time, the program completed successfully. I'm not sure what changed. Perhaps the cloud provider fixed the issue. Should I just close it since I cannot reproduce it?

**Comment by rdyro** [comment]
It's good it's working, yeah, feel free to close it, thanks!
